pkg/gardenlet/controller/shoot/seed_registration_control.go

```diff
@@ -113,884 +68,298 @@ func (c *Controller) seedRegistrationUpdate(oldObj, newObj interface{}) {
-			Domain:     ingressDomain,
-			Controller: *shootedSeedConfig.IngressController,
-		}
-	} else {
-		dns.IngressDomain = &ingressDomain
-	}
-
-	var loadBalancerServices *gardencorev1beta1.SeedSettingLoadBalancerServices
-	if shootedSeedConfig.LoadBalancerServicesAnnotations != nil {
-		loadBalancerServices = &gardencorev1beta1.SeedSettingLoadBalancerServices{
-			Annotations: shootedSeedConfig.LoadBalancerServicesAnnotations,
-		}
-	}
-
-	return &gardencorev1beta1.SeedSpec{
-		Provider: gardencorev1beta1.SeedProvider{
-			Type:           shoot.Spec.Provider.Type,
-			Region:         shoot.Spec.Region,
-			ProviderConfig: shootedSeedConfig.SeedProviderConfig,
-		},
-		DNS:       dns,
-		Ingress:   ingress,
-		SecretRef: secretRef,
-		Networks: gardencorev1beta1.SeedNetworks{
-			Pods:          *shoot.Spec.Networking.Pods,
-			Services:      *shoot.Spec.Networking.Services,
-			Nodes:         shoot.Spec.Networking.Nodes,
-			BlockCIDRs:    shootedSeedConfig.BlockCIDRs,
-			ShootDefaults: shootedSeedConfig.ShootDefaults,
-		},
-		Settings: &gardencorev1beta1.SeedSettings{
-			ExcessCapacityReservation: &gardencorev1beta1.SeedSettingExcessCapacityReservation{
-				Enabled: shootedSeedConfig.DisableCapacityReservation == nil || !*shootedSeedConfig.DisableCapacityReservation,
-			},
-			Scheduling: &gardencorev1beta1.SeedSettingScheduling{
-				Visible: shootedSeedConfig.Visible == nil || *shootedSeedConfig.Visible,
-			},
-			ShootDNS: &gardencorev1beta1.SeedSettingShootDNS{
-				Enabled: shootedSeedConfig.DisableDNS == nil || !*shootedSeedConfig.DisableDNS,
-			},
-			LoadBalancerServices: loadBalancerServices,
-			VerticalPodAutoscaler: &gardencorev1beta1.SeedSettingVerticalPodAutoscaler{
-				Enabled: vpaEnabled,
-			},
+func deleteManagedSeed(ctx context.Context, c client.Client, shoot *gardencorev1beta1.Shoot) error {
+	// Delete managed seed
+	managedSeed := &seedmanagementv1alpha1.ManagedSeed{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      shoot.Name,
+			Namespace: shoot.Namespace,
 		},
-		Taints: taints,
-		Backup: backupProfile,
-		Volume: volume,
-	}, nil
-}
-
-func activateSeedIngressController(ctx context.Context, seedClient client.Client, shoot *gardencorev1beta1.Shoot, ingressController *gardencorev1beta1.IngressController) (bool, error) {
-	if gardencorev1beta1helper.NginxIngressEnabled(shoot.Spec.Addons) {
-		return false, nil
-	}
-
-	if seedIngressEnabled := ingressController != nil; !seedIngressEnabled {
-		return false, nil
 	}
-
-	// If migrating from Shoot Nginx Addon to Seed Ingress Controller wait until DNSEntry pointing to ShootNginxAddon service got deleted before activating Seed Ingress Controller.
-	if err := seedClient.Get(ctx, kutil.Key(shoot.Status.TechnicalID, common.ShootDNSIngressName), &dnsv1alpha1.DNSEntry{}); err != nil {
-		if apierrors.IsNotFound(err) {
-			return true, nil
-		}
-		return false, err
+	if err := c.Delete(ctx, managedSeed); client.IgnoreNotFound(err) != nil {
```

Comment:
Should we consider not support deleting the `ManagedSeed` by removing the `use-as-seed` annotation and rather only support it via a delete call to the `ManagedSeed` resource?
How do you envision the migration path (i.e. to finally get rid of the annotation)?

Reply:
Once the annotation has generated the `ManagedSeed` for the shoot, one could remove the owner ref from the `ManagedSeed` object, and then the annotation itself (without the owner ref this would not delete the `ManagedSeed`). Let me double check that this still works, you could also try it out.

Reply:
Ok, if somebody wants to migrate from the `use-as-seed` annotation those steps would need to be done manually, right? (sounds actually good to me)
Will we then do an automated approach to get rid of all old annotations a few releases into the future?

Reply:
Those steps could also be automated via a script. When you say "automated approach" do you mean this or something different? I am not sure what is the best way to get rid of old annotations, perhaps we should discuss this outside of this PR.

Reply:
Yeah, sure. We'll take care about it later.

Reply:
Tested this, works as expected. When the owner reference is removed, removing the `use-as-seed` annotation no longer deletes the `ManagedSeed` so it stays until deleted manually.